### PR TITLE
net: l2: ethernet: check dev->api and get_capabilities for being NULL

### PR DESCRIPTION
**Pull request by panekmaciej**
Adds missing checks for get_capabilities method not being NULL. Fixes crash with netusb and possibly other drivers.
See #77838

### CODE REVIEW

File: subsys/net/l2/ethernet/ethernet.c [hunk 1]
```diff
@@ -207,29 +204,31 @@ static void ethernet_mcast_monitor_cb(struct net_if *iface, const struct net_add
...
-	if (!(api->get_capabilities(dev) & ETHERNET_HW_FILTERING) || api->set_config == NULL) {
+	if (!(net_eth_get_hw_capabilities(iface) & ETHERNET_HW_FILTERING)) {
 		return;
 	}
 
-	switch (addr->family) {
+	dev = net_if_get_device(iface);
+	api = dev->api;
+
+	if (api && api->set_config) {
```
Comment (jukkar) [inline subsys/net/l2/ethernet/ethernet.c:214]: This change is for set_config callback, could you have a separate commit for it.

File: subsys/net/l2/ethernet/ethernet.c [hunk 2]
```diff
@@ -616,7 +615,7 @@ static int ethernet_send(struct net_if *iface, struct net_pkt *pkt)
 	int ret;
 	struct net_pkt *orig_pkt = pkt;
 
-	if (!api) {
+	if (!api || !api->send) {
```
Comment (jukkar) [inline subsys/net/l2/ethernet/ethernet.c:618]: And this could be placed to separate commit too

**Review by jukkar — COMMENTED** [review]
Looks good but the commit should split to three parts as you are also changing two other callbacks.

File: subsys/net/l2/ethernet/ethernet.c (hunk 1 above)
Comment (panekmaciej) [reply subsys/net/l2/ethernet/ethernet.c]: Done

**Review by jukkar — DISMISSED** [review]

**Review by jukkar — APPROVED** [review]

**Review by pdgendt — APPROVED** [review]

**Review by rlubos — APPROVED** [review]
